xbmc/interfaces/legacy/Addon.h

```diff
@@ -100,10 +100,157 @@ namespace XBMCAddon
+      /// Returns the list of known available instance ids.
+      ///
+      /// @return List of used add-on instances
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      /// @python_v21 New function added.
+      ///
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// usedInstances = self.Addon('pvr.iptvsimple').getKnownInstanceIds()
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      getKnownInstanceIds();
+#else
+      std::vector<unsigned int> getKnownInstanceIds();
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmcaddon
+      /// @brief \python_func{ xbmcaddon.Addon([id]).getFreeNewInstanceId() }
+      /// Get free add-on instance id to set new independent instance and settings about.
+      ///
+      /// @note After setting the new instance settings a @ref setInstanceIdState() call is required.
+      ///
+      /// @return Next usable and free instance id
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      /// @python_v21 New function added.
+      ///
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// addon = xbmcaddon.Addon('pvr.iptvsimple')
+      /// newinstanceid = addon.getFreeNewInstanceId()
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      getFreeNewInstanceId();
+#else
+      unsigned int getFreeNewInstanceId();
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmcaddon
+      /// @brief \python_func{ xbmcaddon.Addon([id]).setInstanceIdState(id, enabled, [name]) }
+      /// Set changed state on add-on instance system.
+      ///
+      /// @param instance unsigned integer - Instance identifier to use.
+      /// @param enabled boolean - To enable or disable the instance
+      /// @param name [opt] string or unicode - The instance name used inside Kodi
+      ///             @warning Needs on new added instances set.
+      /// @return True if the setting the instance change was successful, false otherwise
+      ///
+      /// @note To add or Enable/Disable an instance this function needs to be called.
+      ///
+      ///-----------------------------------------------------------------------
+      /// @python_v21 New function added.
+      ///
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      ///
+      /// addon = xbmcaddon.Addon('pvr.iptvsimple')
+      /// newinstanceid = addon.getFreeNewInstanceId()
+      ///
+      /// settings = addon.getSettings(newinstanceid)
+      /// settings.setInt('m3uPathType', 0)
+      /// settings.setString('m3uPath', '/SOME_PATH_OF_YOU')
+      /// ...
+      /// addon.setInstanceIdState(newinstanceid, True, 'Example by Python added')
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      setInstanceIdState(...);
+#else
+      bool setInstanceIdState(unsigned int instance,
```

Comment:
It seems to me that this function sets the state of an instance, not really of the Id. So preferably: `setInstanceState(Id, value)`

Reply:
Yes, as `setInstanceState(Id, value)` is better and becomes changed.